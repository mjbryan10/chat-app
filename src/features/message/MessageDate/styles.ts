import styled, { ThemeProps, Theme } from 'styled-components';

export const MessageDate = styled.div<ThemeProps<Theme>>`
  max-width: 9em;
  margin: 3px auto;
  padding: 0.5rem;
  border: ${({theme})=> theme.border};
  border-radius: 5px;
  background: ${({theme}) => theme.background.page};
  color: ${({theme}) => theme.color.text ?? 'auto'};
  text-align: center;
`;