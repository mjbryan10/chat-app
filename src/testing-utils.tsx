import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }