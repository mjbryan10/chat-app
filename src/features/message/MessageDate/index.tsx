import React, { FC } from 'react'
import * as S from './styles';

interface Props {
  value: string;
}
const MessageDate: FC<Props> = ({value}) => {
  return (
    <S.MessageDate>
      {value}
    </S.MessageDate>
  )
}

export default MessageDate
