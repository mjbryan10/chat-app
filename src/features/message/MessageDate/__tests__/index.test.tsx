import React from 'react';
import { render } from 'testing-utils';
import MessageDate from '..';

describe('MessageDate Feature Component', () => {
  test('should render the passed down property', () => {
    const testText = "Today";
    const utils = render(<MessageDate value={testText} />)

    const date = utils.getAllByText(/today/i);

    expect(date[0]).toBeVisible();
    expect(date.length).toBe(1);
  })
  
})
