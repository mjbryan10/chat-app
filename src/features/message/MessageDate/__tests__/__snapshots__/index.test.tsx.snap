// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDate Feature Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fTbzEh"
  >
    Today
  </div>
</DocumentFragment>
`;
